### PR TITLE
[TIMOB-11709] Android: String.formatTime causes app crash when date i…

### PR DESCRIPTION
**Pull request by maggieaxway**
…s string

Update TitaniumModule.java

**JIRA:** https://jira.appcelerator.org/browse/TIMOB-11709

```js
var win = Ti.UI.createWindow({
  backgroundColor: 'white'
});

win.addEventListener('click', function() {
  var date = 'GIBBERISH';
  //var date = new Date();
  alert(String.formatTime(date));
  //alert(String.formatDate(date));
});

win.open();
```

### CODE REVIEW

File: android/titanium/src/java/ti/modules/titanium/TitaniumModule.java [hunk 1]
```diff
@@ -274,11 +274,21 @@ public String stringFormatDate(Date date, @Kroll.argument(optional=true) String
 	}
 
 	@Kroll.method @Kroll.topLevel("String.formatTime")
-	public String stringFormatTime(Date time)
+	public String stringFormatTime(Object time)
 	{
 		int style = DateFormat.SHORT;
 
-		return (DateFormat.getTimeInstance(style)).format(time);
+		if (time != null && time instanceof Date) {
```
Comment (jquick-axway) [inline android/titanium/src/java/ti/modules/titanium/TitaniumModule.java]: You don't need to do a `null` check. It's safe to pass a `null` reference to `instanceof` and it'll return false in that case. So, change to this...

```java
if (time instanceof Date) {
```

File: android/titanium/src/java/ti/modules/titanium/TitaniumModule.java [hunk 2]
```diff
@@ -274,11 +274,21 @@ public String stringFormatDate(Date date, @Kroll.argument(optional=true) String
...
+			try {
+				return (DateFormat.getTimeInstance(style)).format(time);
+			} catch (Exception ex) {
+				Log.e(TAG, "Error occurred while formatting time", ex);
+				return null;
+			}
+		}else {
+			Log.e(TAG, "Error occurred while formatting time please provide a Date()");
```
Comment (jquick-axway) [inline android/titanium/src/java/ti/modules/titanium/TitaniumModule.java]: Let's change this error message from...

```java
Log.e(TAG, "Error occurred while formatting time please provide a Date()");
```

...to...

```java
Log.e(TAG, "The string.formatTime() function was given an invalid argument. Must be of type 'Date'.");
```

**Review by jquick-axway — CHANGES_REQUESTED** [review]
@maggieaxway, fixing this on the Java side is definitely simpler. I'm just requesting a couple minor changes.

**Review by jquick-axway — APPROVED** [review]
CR: Pass

**Comment by lokeshchdhry** [comment]
FR Passed.

`String.formatTime("String")` & `String.formatDate("String")` throws error : 
`[ERROR] :  [Nexus 6P] TitaniumModule: (main) [2678,2678] The string.formatTime() function was given an invalid argument. Must be of type 'Date'.
[ERROR] :  [Nexus 6P] TitaniumModule: (main) [38,2716] The string.formatDate() function was given an invalid argument. Must be of type 'Date'.`

No app crash seen.

Studio Ver:     4.10.0.201709271713
SDK Ver:        7.0.0 local build
OS Ver:         10.12.3
Xcode Ver:      Xcode 8.3.3
Appc NPM:       4.2.10
Appc CLI:       6.3.0
Ti CLI Ver:     5.0.14
Alloy Ver:      1.10.7
Node Ver:       7.10.1
Java Ver:       1.8.0_101
Devices:        ⇨ google Nexus 5 --- Android 6.0.1
                ⇨ google Nexus 6P --- Android 8.0.0

**Pull request merged**